sdk/client/src/broadcast_client.rs

```diff
@@ -0,0 +1,216 @@
+            .map(|client| client.get_events(key, start_seq, limit));
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_currencies(&self) -> Result<Vec<Response<Vec<CurrencyInfoView>>>> {
+        let futures = self.random_clients().map(|client| client.get_currencies());
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_network_status(&self) -> Result<Vec<Response<u64>>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_network_status());
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    //
+    // Experimental APIs
+    //
+
+    pub async fn get_state_proof(&self, from_version: u64) -> Result<Vec<Response<StateProofView>>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_state_proof(from_version));
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_account_state_with_proof(&self,
+                                              address: AccountAddress,
+                                              from_version: Option<u64>,
+                                              to_version: Option<u64>, ) -> Result<Vec<Response<AccountStateWithProofView>>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_account_state_with_proof(address,
+                                                              from_version,
+                                                              to_version, ));
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_transactions_with_proofs(&self,
+                                              start_version: u64,
+                                              limit: u64, ) -> Result<Vec<Response<()>>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_transactions_with_proofs(start_version,
+                                                              limit, ));
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_events_with_proofs(&self,
+                                        key: &str,
+                                        start_seq: u64,
+                                        limit: u64, ) -> Result<Vec<Response<()>>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_events_with_proofs(key,
+                                                        start_seq,
+                                                        limit, ));
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+}
+
+fn collect_results<T>(mut results: Vec<Result<Response<T>>>) -> Result<Vec<Response<T>>> {
```

Comment:
Take a look at the old `select_highest_ledger_version_response` for how to select just 1 of the N responses to return.